###2021年6月29日23:02:42

	opensource physics engine
		https://3g2upl4pq6kufc4m.onion/?q=opensource+physics+engine&ia=web
	python physics engine
		https://3g2upl4pq6kufc4m.onion/?q=python+physics+engine&ia=weby
		https://projectchrono.org/pychrono/	###id=BA72

###2021年6月30日10:21:09

>>
	https://projectchrono.org/pychrono/
	1. install https://www.anaconda.com/
		https://www.anaconda.com/products/individual
		1. dl
			https://repo.anaconda.com/archive/Anaconda3-2021.05-Windows-x86_64.exe
		2 install	[12:05:32
		
	2. tut
		https://docs.anaconda.com/anaconda/user-guide/?utm_source=anaconda.com&utm_medium=individual-get-started
		
		##id=LS5U https://docs.anaconda.com/anaconda/user-guide/getting-started/	<== this : 2021年7月4日14:34:44	
		
		#1
		#2 Spyder by clicking Spyder’s
			PROBLEM		[12:52:10
				--> spyder, console opened, not starting
			TRY		[12:52:54
				1. restart navigator
				2. launch Spyder
					--> splash started	[12:53:28
					--> IDE ==> started		[12:54:08
		#3 In the new file on the left,
			?? "placeholder text"
		#4 In the top menu, click File 
			C:\Users\iwabuchiken\.spyder-py3\hello.py
		#5 Close Spyder		[12:59:50
	n:
		#6 What’s next?


###2021年7月4日14:32:00

>>
	*start --> anaconda navigator
		--> start from "start" button
		]14:38:53
		
	1. browse --> -->##id=LS5U
	-->##id=LS5U https://docs.anaconda.com/anaconda/user-guide/getting-started/
	
		#1 Run Python in a Jupyter Notebook
			1. launch --> click
				==> OK		]14:41:46
		*#2 On the top of the right hand side	[14:42:10
			1. save notebook
				==> OK		]14:47:07
		*#3 Close Jupyter Notebook
			]14:57:34
	
	2. physics engine : -->###id=BA72
		https://api.projectchrono.org/pychrono_introduction.html	PyChrono
		https://api.projectchrono.org/pychrono_installation.html	Install PyChrono
			#1 Alternatively, use the Anaconda Navigator GUI, 		[15:11:56
				1. anaconda navi
					1. button "Channels" --> click
					2. new window : "Add"
					3. "projectchrono" --> channel name
						search
							pychrono channel name anaconda 
					4. navi : "refresh"
		SEG-1
			1. search
				pychrono インストール
					==> no relevant
				pychrono 使い方
					==> no relevant
				pychrono anaconda gui install
			*TRY		[15:26:08
					re-start navi
				1. ==> n/c		]15:28:21
			TRY : search
				anaconda navi channels
				1. https://anaconda.org/search?q=pychrono
				2, "https://anaconda.org/projectchrono"	==> add channel
				==> n/c (in the navi, main pane		]15:33:43
				--> pychrono channel url : https://anaconda.org/projectchrono
					--> ref doc : https://docs.anaconda.com/anaconda/navigator/tutorials/manage-channels/ "EXAMPLE: Let’s say you want to "
			TRY
				navi, left pane --> "Environ..."
				--> "pychrono" -> items exists
			TRY		[15:35:51
				install
				1. click the box button
				2. "Add" button --> click	[15:36:49
				3. dialog : "Apply" --> click	[15:38:08
				
		n:		[15:53:39
		SEG-2	verify : jupyter, import pychrono	--> as in https://api.projectchrono.org/pychrono_introduction.html "The idea is that, once installed, "		
				
	SEG-3 : unrelated
		https://qiita.com/search?q=シミュレーション
		https://qiita.com/Teeca/items/0268aa23d1abec189199
		https://qiita.com/hiRina/items/0ae29a679a6c01dc8aab

###2021年7月4日16:33:03		s-2

>>>
	SEG-2	verify : jupyter, import pychrono	--> as in https://api.projectchrono.org/pychrono_introduction.html "The idea is that, once installed, "		
		1. jupyter --> start
			file:///C:/Users/iwabuchiken/AppData/Roaming/jupyter/runtime/nbserver-8784-open.html
			http://localhost:8888/tree?token=c471f2ee204771b08ba6f90f4ccd53bcc12688d7c5e322b7
		2. "new" > "Python 3"
		3. console
			1. "import pychrono as chrono"
			ERROR
				"ImportError: DLL load failed with error code 3221225501 while importing _core"
			TRY : search
				pychrono ImportError DLL load failed
			TRY : browse file
				1. pushd C:\WORKS_2\Programs\anaconda3_2021\lib\site-packages\pychrono\
				C:\WORKS_2\Programs\anaconda3_2021\lib\site-packages\pychrono\__init__.py
				*2. DUP : file		]16:43:08
				3. search : "_core.py" in C:\WORKS_2\Programs\anaconda3_2021	[16:45:14
					--> _core.pyd // C:\WORKS_2\Programs\anaconda3_2021\pkgs\pychrono-6.0.0-py38_0\Lib\site-packages\pychrono
			TRY : browse file
				core.py // C:\WORKS_2\Programs\anaconda3_2021\lib\site-packages\pychrono\core.py
				//sakura C:\WORKS_2\Programs\anaconda3_2021\lib\site-packages\pychrono\core.py
				--> _core.pyd // C:\WORKS_2\Programs\anaconda3_2021\Lib\site-packages\pychrono
			*TRY : show PYTHONPATH val		]17:08:14
				*0. test : in core.py, write a non-significant code		]17:01:26
					1. coding : "print("yes")"	--> #debug:20210704_165804
					2. jupyter --> again, import chrono : "import pychrono as chrono"
					==> ERROR : "IndentationError: unindent does not " --> OK. the control is loading the debug line
					3. fix the code, re-run
					==> "yes" --> shown in the error message chunk.
				*1. search : python show PYTHONPATH value
					https://stackoverflow.com/questions/1489599/how-do-i-find-out-my-pythonpath-using-python
					*1. TRY : import sys; print(sys.path)
						1. coding
						2. TEST ==> shown. OK		]17:04:30
					*2. pychrono lib --> in the PATH ?
						==> YES --> NO
						C:\WORKS_2\Programs\anaconda3_2021\Lib\site-packages\pychrono --> no
						C:\WORKS_2\Programs\anaconda3_2021\Lib\site-packages --> no
			TRY : add pychrono path		[17:08:33
				1. search : python add lib path
					https://stackoverflow.com/questions/20394613/add-a-folder-to-the-python-library-path-once-for-all-windows		<== use "sys"
					https://stackoverflow.com/questions/23244418/set-ld-library-path-before-importing-in-python		<== use "os"
				2. coding
					*1. debug lines --> c/o	(-->#debug:20210704_165804)
					*2. add a line : #code:20210704_171218
						//C:\WORKS_2\Programs\anaconda3_2021\lib\site-packages\pychrono
						"sys.path.append('C:\\WORKS_2\\Programs\\anaconda3_2021\\lib\\site-packages\\pychrono')"
				3. test
					*1. execution-code-less lines ==> OK		]17:18:36
//SEG-20210704_171942
//	1. if this test "2" shows no success --> halt the pychrono. start qiita articles
// maybe, for using pychrono, use conda through console.	17:31:57

					2. execution-code-added lines
						*1. c/i the line		]17:20:33
						2. jupyter --> exec	
					ERROR : NameError: name 'sys' is not defined
				*TRY : "import sys"
					1. add line at : #code:20210704_172216
					2. test
						==> ERROR : "ImportError: DLL load failed with error code 3221225501 while importing _core"
				*TRY : after appending the path, re-show the PATH val
					1. coding
					2. test 
						==> n/c		]17:26:09
//['C:\\Users\\iwabuchiken', 'C:\\WORKS_2\\Programs\\anaconda3_2021\\python38.zip', 'C:\\WORKS_2\\Programs\\anaconda3_2021\\DLLs', 'C:\\WORKS_2\\Programs\\anaconda3_2021\\lib', 'C:\\WORKS_2\\Programs\\anaconda3_2021', '', 'C:\\WORKS_2\\Programs\\anaconda3_2021\\lib\\site-packages', 'C:\\WORKS_2\\Programs\\anaconda3_2021\\lib\\site-packages\\locket-0.2.1-py3.8.egg', 'C:\\WORKS_2\\Programs\\anaconda3_2021\\lib\\site-packages\\win32', 'C:\\WORKS_2\\Programs\\anaconda3_2021\\lib\\site-packages\\win32\\lib', 'C:\\WORKS_2\\Programs\\anaconda3_2021\\lib\\site-packages\\Pythonwin', 'C:\\WORKS_2\\Programs\\anaconda3_2021\\lib\\site-packages\\IPython\\extensions', 'C:\\Users\\iwabuchiken\\.ipython', 'C:\\WORKS_2\\Programs\\anaconda3_2021\\lib\\site-packages\\pychrono', 'C:\\WORKS_2\\Programs\\anaconda3_2021\\lib\\site-packages\\pychrono']						
				*search : python pyd file	]17:29:35
					==> what is pyd file?
					https://stackoverflow.com/questions/8822335/what-do-the-python-file-extensions-pyc-pyd-pyo-stand-for#8822377
				
				search		<== this is the last try
					pychrono import dll file failed
					https://github.com/projectchrono/chrono/issues/140
					https://groups.google.com/g/projectchrono/c/LCogsynC7VM		##d:20210704_173453
	n:
		this time -->##d:20210704_173453
		1. maybe, you need an actual DLL file itself (such as ***.dll), not a pyd file.

]17:35:30

	1. after-closing works	[17:51:56
		1. freemind --> register this project : prog > physic-engine
			*1. new file : 57_programming
		2. git
			*1. copy command line strings --> from : 72. read
			*2. exec git

###2021年7月5日9:28:53
s-3 / g-2

>>
	1. start --> 
		*1. navi		]10:48:01
		2. jupyter
		
	*2. re-make-it-happen : error case		]10:51:42
			"import pychrono as chrono"
		1. "new" > Python 3
		2. paste : line --> run (S - Ret
			==> "ImportError: DLL load failed with error code 3221225501 while importing _core"
	3.TRY : -->##d:20210704_173453
			https://groups.google.com/g/projectchrono/c/LCogsynC7VM
		#1 
		#2 i check on my directory;
			// review : _core.pyd // C:\WORKS_2\Programs\anaconda3_2021\Lib\site-packages\pychrono
			SEG-1 : search dll
				--> ChronoEngine.dll @C:\WORKS_2\Programs\anaconda3_2021
				1. FOUND
					1. C:\WORKS_2\Programs\anaconda3_2021\DLLs --> 
						1. no -->  "_core.pyd"
						2. ChronoEngine.dll --> no
					2. C:\WORKS_2\Programs\anaconda3_2021\bin
						==> no
					3. C:\WORKS_2\Programs\anaconda3_2021\pkgs\pychrono-6.0.0-py38_0\Library\bin
						1. ChronoEngine.dll --> YES
			*SEG-2 : how to? --> add dll-load path
				1. search : python path dll
					https://stackoverflow.com/questions/2980479/python-ctypes-loading-dll-from-from-a-relative-path "I always add the directory"
				XX. conc
					1. add the path to the "...\library\bin"
						C:\WORKS_2\Programs\anaconda3_2021\Library\bin
						--> one of the folder where the dll exists.//DLL이 존재하는 폴더의 하나.
			SEG-3 : add path to dll file
				1. core.py -->#debug:20210704_165804
					C:\WORKS_2\Programs\anaconda3_2021\lib\site-packages\pychrono
					*[[open file : sakura C:\WORKS_2\Programs\anaconda3_2021\lib\site-packages\pychrono\core.py
				[[*2. coding	[11:21:01
					#code:20210705_112130
				3. test --> run		[11:24:53
					==> n/c		]11:26:14
		#3 Have you also installed the pre-compiled windows binaries: 	[11:28:10
		#4 managed to borrow a more up-to-date laptop, 
	4. conc
		1. seems the computer is not up for pychrono.	]11:41:59

>>> [11:42:30
	1. search : python physics engine
		https://3g2upl4pq6kufc4m.onion/?q=python+physics+engine&ia=web
		https://docs.panda3d.org/1.10/python/programming/physics/builtin/index
	
	2. search :
			python pandas tutorial
			python panda3d phsics engine
			python panda3d tutorial

n:
	panda3d
	https://docs.panda3d.org/1.10/python/introduction/index

]12:14:13

###17:38:23


>>> read

	panda3d
	https://docs.panda3d.org/1.10/python/introduction/index
		C:\WORKS_2\WS\WS_Others.JVEMV6\JVEMV6\70_co2\texts-for-reading\out.20210705_174008_960514.txt
			scene graph browsing
		*C:\WORKS_2\WS\WS_Others.JVEMV6\JVEMV6\70_co2\texts-for-reading\out.20210705_174641_498166.txt		]17:53:50
		
n:
	https://docs.panda3d.org/1.10/python/introduction/installation-windows
	Installing Panda3D in Windows

]17:54:26

###2021年7月6日17:02:21


	https://docs.panda3d.org/1.10/python/introduction/installation-windows
	Installing Panda3D in Windows
		C:\WORKS_2\WS\WS_Others.JVEMV6\JVEMV6\70_co2\texts-for-reading\out.20210706_171056_015088.txt
		#1
			https://www.panda3d.org/download/
			https://www.panda3d.org/download/sdk-1-10-9/
			pip install panda3d==1.10.9
			*python -m pip list
			*python -m pip install panda3d==1.10.9
//C:\WORKS_2>python -m pip install panda3d==1.10.9
//Collecting panda3d==1.10.9
//  Downloading https://files.pythonhosted.org/packages/36/95/67c819f25986e1c98660
//7aa93656f5a4a59b53efff082571b53e1c5c9c5d/panda3d-1.10.9-cp35-cp35m-win_amd64.whl
// (61.7MB)
//    100% |################################| 61.7MB 16kB/s
//Installing collected packages: panda3d
//Successfully installed panda3d-1.10.9
//You are using pip version 9.0.1, however version 21.1.3 is available.
//You should consider upgrading via the 'python -m pip install --upgrade pip' comm
//and.
			*python -m pip list	[17:20:29
				==> listed		]17:20:56
		#2 run a sample program
			C:\WORKS_2\Programs\Python\Python_3.5.1\
			==> PROBLEM
				--> "samples" folder ~~> not in the tree
			TRY : use installer : https://www.panda3d.org/download/sdk-1-10-9/	[17:32:17
				"Installer for Windows (64-bit) "
				*1. download	]17:42:00
				*2. run exe file
					C:\WORKS_2\Programs\panda3d\Panda3D-1.10.9-x64
				3. samples folder
					pushd C:\WORKS_2\Programs\panda3d\Panda3D-1.10.9-x64\samples
					pushd fractal-plants
					python main.py

n:
	#n:20210706_180605
	C:\WORKS_2\WS\WS_Others.JVEMV6\JVEMV6\70_co2\texts-for-reading\out.20210706_171056_015088.txt

###2021年7月7日10:51:23


>>>

#n:20210706_180605
	C:\WORKS_2\WS\WS_Others.JVEMV6\JVEMV6\70_co2\texts-for-reading\out.20210706_171056_015088.txt
	
	pushd C:\WORKS_2\Programs\panda3d\Panda3D-1.10.9-x64\samples
 ドライブ C のボリューム ラベルがありません。
 ボリューム シリアル番号は 84A5-EBF6 です

 C:\WORKS_2\Programs\panda3d\Panda3D-1.10.9-x64\samples のディレクトリ

set name=rocket-console
echo dir\%name%"
dir %name%
%name%\main.py
%name%\basic.py
%name%\advanced.py


// 2021/07/06  17:45    <DIR>          .
// 2021/07/06  17:45    <DIR>          ..
// 2021/07/06  17:45    <DIR>          asteroids
// 2021/07/06  17:45    <DIR>          ball-in-maze
// 2021/07/06  17:45    <DIR>          boxing-robots
// 2021/07/06  17:45    <DIR>          bump-mapping		#
// 2021/07/06  17:45    <DIR>          carousel			#
// 2021/07/06  17:45    <DIR>          cartoon-shader
// 2021/07/06  17:45    <DIR>          chessboard
// 2021/07/06  17:45    <DIR>          culling			#
// 2021/07/06  17:45    <DIR>          disco-lights
// 2021/07/06  17:45    <DIR>          distortion		#
// 2021/07/06  17:45    <DIR>          fireflies		#
// 2021/07/06  17:45    <DIR>          fractal-plants	#
// 2021/07/06  17:45    <DIR>          gamepad
// 2021/07/06  17:45    <DIR>          glow-filter		#
// 2021/07/06  17:45    <DIR>          infinite-tunnel	#
// 2021/07/06  17:45    <DIR>          looking-and-gripping	#
// 2021/07/06  17:45    <DIR>          media-player		***
// 2021/07/06  17:45    <DIR>          motion-trails	#
// 2021/07/06  17:45    <DIR>          mouse-modes
// 2021/07/06  17:45    <DIR>          music-box
// 2021/07/06  17:45    <DIR>          particles
// 2021/07/06  17:45    <DIR>          procedural-cube
// 2021/07/06  17:45    <DIR>          render-to-texture
// 2021/07/06  17:45    <DIR>          roaming-ralph
// 2021/07/06  17:45    <DIR>          rocket-console
// 2021/07/06  17:45    <DIR>          shader-terrain
// 2021/07/06  17:45    <DIR>          shadows			#
// 2021/07/06  17:45    <DIR>          solar-system		#
//                0 個のファイル                   0 バイト
//               30 個のディレクトリ  37,398,331,392 バイトの空き領域

n:
	https://docs.panda3d.org/1.10/python/introduction/running-your-program
	Running your Program

]11:44:57


###2021年7月7日16:57:12


	https://docs.panda3d.org/1.10/python/introduction/running-your-program
	Running your Program
	
		C:\WORKS_2\WS\WS_Others.JVEMV6\JVEMV6\70_co2\texts-for-reading\out.20210707_165703_232743.txt
		#1
			cd C:\WORKS_2\Programs\panda3d\Panda3D-1.10.9-x64

		#2 To run your script and start
		ppython myscript.py
		==> DROP : "myscript.py" --> where?

	https://docs.panda3d.org/1.10/python/introduction/tutorial/index
	A Panda3D Hello World Tutorial
	
	#1 This tutorial is called 
		C:\WORKS_2\WS\WS_Others.JVEMV6\JVEMV6\70_co2\texts-for-reading\out.20210707_172205_284254.txt
	SEG-1
		file : C:\WORKS_2\WS\WS_Others.JVEMV6\JVEMV6\57.2_physics-engine\s-1\s_1.py
		1. run file
			ERROR : "ModuleNotFoundError: No module named 'sympy'"
			TRY : add path
				#code:20210707_175722
				==> n/c		]17:58:11
			WORKAROUND : c/o "sympy"
			
n:
	C:/WORKS_2/WS/WS_Others/free/fx/82_\libs\libs.py
		==> DUP. make a new file "libs_20210707_180120.py" --> when in pada3d, let's use this file

###
@@@




###





###





###2021年7月5日12:15:02
====================== git

rem set stringof_session_id="s-6 / g-2" 

C:\WORKS_2\start_a.bat

set stringof_project_id="JVE_57.[2_physics-engine]" 
C:\WORKS_2\shortcuts_docs\end_log_JVE_ALL.[ai].git.bat %stringof_project_id:"=%



###2021年7月7日10:59:55
comment out

^(.)	// $1


###2021年7月5日17:56:26
====================== closing

C:\WORKS_2\shortcuts_docs\end_log_JVE_close-sakura_close-freemind.bat


"C:\WORKS_2\shortcuts_docs\end_log_JVE_70.[CO2]_apps.bat"

